source/slang-wasm/slang-wasm.cpp

```diff
@@ -386,6 +386,47 @@ HashedString* ComponentType::loadStrings()
     return hashedStrings;
 }
 
+ProgramLayout* ComponentType::getLayout(unsigned int targetIndex)
```

Comment:
This is going to making really hard in javascript to ensure all these reflection objects are explicitly `delete()`'d when they are no longer in use. 

We should instead just return a raw pointer to the layout without allocating a fresh object on the heap, and use `allow_raw_pointers` in emscripten bind, to free the javascript side from the responsibility of manually calling delete().

Reply:
Okay, let me see if I can change up the wasm bindings